### PR TITLE
Allow filterable search by date or alphabetically

### PR DESCRIPTION
**Pull request by chosak**
This change fixes an existing bug in how filterable list search doesn't always properly order search results by date (internal
D&CP#234). It also adds the ability to order search results alphabetically by the title of the filtered pages (which addresses
internal D&CP#238).

With this change every page on the site will be using order by date.

This logic works by adding two new fields to the search index: `start_date` and `end_date` which are used for all filtering and ordering. Previously, we indexed fields from our different page types (like EventPage's `start_dt` or EnforcementActionPage's `initial_filing_date`) and then had to configure the search to use the right field for filtering/ordering. This way, we always use consistent fields for those operations and only have to make sure that the correct
page model fields get indexed at index time.

This change does remove the option to sort results by search "score" - this isn't used publicly on the site anywhere and hasn't been tested. We retain the ability to re-add this at some point in the future.

## How to test this PR

To test, either run a full refresh-data.sh or re-index your local search index with:

```
./cfgov/manage.py opensearch index --force rebuild
./cfgov/manage.py opensearch document --force --refresh --parallel index
```

Then try editing a page with a FilterableList and try changing the sort order to alphabetical.

## Screenshots

imageimage

## Notes and todos

 I've chosen not to add a data migration to migrate existing fields in the interest of simplicity. We will need to run manual search index reindexing to activate this once it gets deployed.

## Checklist

- [x] PR has an informative and human-readable title
- [x] Changes are limited to a single goal (no scope creep)
- [x] Code follows the standards laid out in the [CFPB development guidelines](https://github.com/cfpb/development)

### CODE REVIEW

File: cfgov/v1/documents.py [hunk 1]
```diff
@@ -152,11 +152,20 @@ def filter_language(self, language=None):
...
+            ranges = [
+                {"end_date": {"gte": from_date}},
+                {"start_date": {"lte": to_date}},
+            ]
+        elif from_date:
+            ranges = [{"end_date": {"gte": from_date}}]
+        elif to_date:
+            ranges = [{"start_date": {"lte": to_date}}]
+        else:
+            ranges = []
```
Comment (chosak) [inline cfgov/v1/documents.py:165]: @csebianlander I tried to write tests for this but I would appreciate your once-over on this logic to make sure I did this right. What I tried to do here was make the ranges inclusive; in other words, say I have an event that starts at 12PM and ends at 1PM. If I search with a range from START to END, then END has to be after 12PM, and start has to be before 1PM. If I give only START, then START has to be before 1PM (my range overlaps with the end of the event) or, if I give only END, then END has to be after 12PM (my range overlaps with the beginning of the event). Does this seem right to you?

Comment (csebianlander) [reply cfgov/v1/documents.py:165]: yeah that seems right to me!

**Comment by csebianlander** [comment]
Ordering seems correct (date and alpha). I'm having some problems with my local environment that are probably unrelated but I thought I'd share -- I can't expand the filter tools and it looks like this:

Screen Shot 2023-05-15 at 2 15 50 PM

**Comment by chosak** [comment]
> I'm having some problems with my local environment

Have you rebuilt the frontend lately? (`./frontend.sh`) The events page filter expands/closes for me.

**Comment by csebianlander** [comment]
I have, but no luck. In any case the functionality described here works exactly and I assume the other thing is my own problem!

**Review by csebianlander — APPROVED** [review]

**Pull request merged**